### PR TITLE
Fix: nil check for Proxy IpAddress

### PR DESCRIPTION
**Pull request by yangdihangN**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
when is it nil?

**Comment by hzxuzhonghu** [comment]
We can prevent it here, which is more direct.

```
	// Does query from ingress or router have to carry valid IP address?
	if len(out.IPAddresses) == 0 && out.Type == SidecarProxy {
		return out, fmt.Errorf("no valid IP address in the service node id or metadata")
	}
```

File: pilot/pkg/serviceregistry/kube/controller.go [hunk 1]
```diff
@@ -499,31 +499,34 @@ func (c *Controller) InstancesByPort(hostname model.Hostname, reqSvcPort int,
...
-	pod := c.pods.getPodByIP(proxyIP)
-	if pod != nil {
-		if !pilot.DisableSplitHorizonEdsProxyNetworkCompare() {
-			// for split horizon EDS k8s multi cluster, in case there are pods of the same ip across clusters,
-			// which can happen when multi clusters using same pod cidr.
-			// As we have proxy Network meta, compare it with the network which endpoint belongs to,
-			// if they are not same, ignore the pod, because the pod is in another cluster.
-			if proxy.Metadata[model.NodeMetadataNetwork] != c.endpointNetwork(proxyIP) {
-				return out, nil
+	if proxy.IPAddresses != nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller.go]: len(proxy.IPAddresses) > 0 ?

Comment (yangdihangN) [reply pilot/pkg/serviceregistry/kube/controller.go]: yes,  it is more strict, thanks

**Comment by hzxuzhonghu** [comment]
@yangdihangN Seems your repo is too old, we have removed circleci. So you should rebase your branch. 

/lgtm

**Comment by howardjohn** [comment]
submit to master branch not release-1.1

**Comment by yangdihangN** [comment]
@googlebot I consent.

**Pull request merged**